source/hash_table.c

```diff
@@ -496,6 +496,38 @@ int aws_hash_table_create(
     return AWS_OP_SUCCESS;
 }
 
+AWS_COMMON_API
+int aws_hash_table_put(struct aws_hash_table *map, const void *key, void *value, int *was_created) {
+    struct hash_table_state *state = map->p_impl;
+    struct aws_hash_element *p_elem;
+    int was_created_fallback;
+
+    if (!was_created) {
+        was_created = &was_created_fallback;
+    }
+
+    /* We might end up resizing the table, so stash the destroy fns now */
+    aws_hash_element_destroy_fn *destroy_key_fn = state->destroy_key_fn;
+    aws_hash_element_destroy_fn *destroy_value_fn = state->destroy_value_fn;
+
+    if (aws_hash_table_create(map, key, &p_elem, was_created)) {
+        return AWS_OP_ERR;
+    }
+
+    if (!*was_created) {
+        if (p_elem->key != key) {
+            destroy_key_fn((void *)p_elem->key);
+        }
+
+        destroy_value_fn((void *)p_elem->value);
+    }
+
+    p_elem->key = key;
```

Comment:
Very minor, but strictly this only needs to be inside the conditional started at line 517. Otherwise it's a redundant assignment.

Reply:
True enough, but it's basically free and more clearly captures the intent.